garden_ai/hpc/groundhog.py

```diff
@@ -21,18 +22,20 @@ def load_function_from_source(contents: str, name) -> Function:
         temp_file.write(contents)
         script_path = Path(temp_file.name)
         with groundhog_script_path(script_path):
-            import __main__
-
-            # exec the script to init the Function instance
-            exec(contents, __main__.__dict__)
-
-            obj = __main__.__dict__.get(name)
-            if obj is None:
-                raise ValueError(f"No groundhog function {name} found in script")
-            elif not isinstance(obj, Function):
-                raise ValueError(
-                    f"Expected {name} to be a groundhog function, got: {type(obj)}"
-                )
+            module = import_user_script(f"{name}_module", script_path)
+            obj = module.__dict__.get(name)
+            # import __main__
+            #
```

Comment:
This seems like a functionality change and not a schema change? (I am not deep enough in recent developments to tell if this makes sense or not, sadly)